src/h3lib/include/h3api.h.in

```diff
@@ -734,6 +735,17 @@ DECLSPEC H3Error H3_EXPORT(experimentalLocalIjToH3)(H3Index origin,
                                                     H3Index *out);
 /** @} */
 
+int isLow52Sorted(const H3Index *cells, const int64_t N);
+H3Error low52Sort(H3Index *cells, const int64_t N);
+
+int isCanonicalCells(const H3Index *cells, const int64_t N);
```

Comment:
Ugh. Maybe `isCanonicalSet` or `isOrderedSet`?

Basically I want singular, type-style names (and maybe type aliases) for the different kinds of arrays, e.g. `OrderedList`, `OrderedSet`, `CompactOrderedSet`. Maybe these should even be structs with a "type" enum specifying which one they are, so that set operations can work on all of them equivalently.

Reply:
I like this idea! I think things like `OrderedList`, `OrderedSet`, `CompactOrderedSet` would definitely help to make things clearer.

But that starts to suggest changes to the API to start using these structs instead of our usual pointer and number of elements. Would we want to do that?

Reply:
Also, I was thinking of using "canonical" instead of "set", because they're slightly different. You can have a distinct **set** of cells that includes parent/child cells, but canonical forbids that. So wanted a concept that was very obvious to a reader as distinct from the usual notion of "set".

What do you think?

Reply:
What about something like `Low52List`, `CanonicalSet`? I'm not yet sure if we need a separate type for compact and canonical tho...

Reply:
See my comment above. I think having these new functions (and, for the moment, only these new functions) use the structs as semi-opaque objects would be valuable, and I would choose names that indicate the use cases and tradeoffs the user incurs, rather than the implementation that leads to these tradeoffs.